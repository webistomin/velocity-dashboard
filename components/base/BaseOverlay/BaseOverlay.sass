+b('base-overlay')
  position: fixed
  top: 0
  left: 0
  width: 100vw
  height: 100%
  background-color: rgb(var(--color-gray-1))
  opacity: 0
  visibility: hidden
  transition: opacity 0.2s ease-out, visibility 0.2s ease-out
  pointer-events: none

  +m('visible')
    opacity: 0.5
    visibility: visible
    pointer-events: all
