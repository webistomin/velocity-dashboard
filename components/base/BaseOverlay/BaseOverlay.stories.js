import { boolean } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import BaseOverlay from './BaseOverlay';

export default {
  title: 'BaseOverlay',
  parameters: {
    options: {
      showPanel: true,
      selectedPanel: 'storybook/knobs/panel',
    },
  },
};

export const Default = () => ({
  components: { BaseOverlay },
  props: {
    isVisible: {
      type: Boolean,
      default: boolean('isVisible', true),
    },
  },
  methods: {
    onClick: action('emit click'),
  },
  template: `<base-overlay
    :isVisible='this.isVisible'
    @click='this.onClick'
  />`,
});
