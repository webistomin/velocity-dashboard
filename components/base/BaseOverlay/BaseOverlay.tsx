import { VueComponent } from 'types/vue-components';
import { Component, Emit, Prop } from 'nuxt-property-decorator';
import { VNode } from 'vue';

import './BaseOverlay.sass';

export interface IBaseOverlayProps {
  isVisible: boolean;
  onClick?: () => void;
}

@Component({
  name: 'BaseOverlay',
})
export default class BaseOverlay extends VueComponent<IBaseOverlayProps> {
  @Prop({ default: false })
  private readonly isVisible!: IBaseOverlayProps['isVisible'];

  @Emit('click')
  onOverlayClick(): void {}

  render(): VNode {
    return <div class={`base-overlay ${this.isVisible ? 'base-overlay_visible' : ''}`} onClick={this.onOverlayClick} />;
  }
}
