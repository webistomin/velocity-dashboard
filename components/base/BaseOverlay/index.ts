import BaseOverlay from './BaseOverlay';

export default BaseOverlay;
