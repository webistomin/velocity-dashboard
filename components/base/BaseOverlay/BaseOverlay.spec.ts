import { mount } from '@vue/test-utils';
import BaseOverlay from './BaseOverlay';

describe('BaseOverlay', () => {
  const overlaySelector = '.base-overlay';

  const factory = (propsData = {}, options = {}) => {
    return mount(BaseOverlay, {
      propsData: {
        isVisible: false,
        ...propsData,
      },
      ...options,
    });
  };

  it('Matches snapshot', () => {
    const wrapper = factory().html();
    expect(wrapper).toMatchSnapshot();
  });

  it('Emit "click" event', () => {
    const wrapper = factory();
    const input = wrapper.find(overlaySelector);
    input.trigger('click');
    expect(wrapper.emitted().click);
  });

  it('Set visible class if color visible is not provided', () => {
    const expectedResult = ['base-overlay_visible'];
    const wrapper = factory({
      isVisible: true,
    });
    const classes = wrapper.classes();
    expect(classes).toEqual(expect.arrayContaining(expectedResult));
  });
});
