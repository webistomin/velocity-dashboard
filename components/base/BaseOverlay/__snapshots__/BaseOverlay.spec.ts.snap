// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseOverlay Matches snapshot 1`] = `"<div class=\\"base-overlay \\"></div>"`;
